server/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -773,22 +691,32 @@ public DoubleArray grow(DoubleArray array, long minSize) {
         return resize(array, newSize);
     }
 
-    public static class DoubleBinarySearcher extends BinarySearcher<Double>{
+    public static class DoubleBinarySearcher extends BinarySearcher{
```

Comment:
Its probably worth a note that this class isn't thread safe. Just to cover all your bases.